lib/maven/build/dockerizedMavenBuilder.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    ChildProcessResult,
+    RemoteRepoRef,
+    spawnAndWatch,
+} from "@atomist/automation-client";
+import {
+    AppInfo,
+    GitProject,
+    LocalProject,
+    ProgressLog,
+} from "@atomist/sdm";
+import {
+    Builder,
+    BuildInProgress,
+} from "@atomist/sdm-pack-build";
+import { MavenProjectIdentifier } from "../parse/pomParser";
+import { VersionedArtifact } from "../VersionedArtifact";
+
+/* tslint:disable:max-classes-per-file */
+
+export const DefaultMavenDockerVersion = "3.5-jdk-8-alpine";
+
+/**
+ * Build with Maven in the local automation client.
+ * This implementation uses Docker and does not require a Java or Maven installation.
+ * Note it is NOT intended for use for multiple organizations. It's OK
+ * for one organization to use inside its firewall, but there is potential
+ * vulnerability in builds of unrelated tenants getting at each others
+ * artifacts.
```

Comment:
Is that comment still accurate? Each build runs in its own container, how can artifacts be shared in that case?